docs/types.rst

```diff
@@ -857,13 +859,26 @@ or create a new memory array and copy every element.
         }
     }
 
-.. index:: ! array;literals, !inline;arrays
+.. index:: ! array;literals, ! inline;arrays
+
+Array Literals
+^^^^^^^^^^^^^^
+
+Array literals (also called inline arrays) are arrays that are written as an
+expression and are not assigned to a variable right away.
+
+::
+
+  pragma solidity >=0.4.16 <0.6.0;
 
-Array Literals / Inline Arrays
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  contract C {
+    function f() public pure {
+      uint[] memory a;
```

Comment:
This is not an inline array. It is a declaration of a variable of array type. If you want to have a very simple example that is different from the one below, it might be better to just use a snippet and not a full contract. Something like `[1, x, y]`.

Reply:
@chriseth OK, I must admit, I found the opening paragraph super confusing, and it was leading to my confusion and continual wrong examples. Unless I have completely miss-understood something (and I read some other language docs to clarify), I changed that opening paragraph (paraphrased from the JS MDN docs), and then the example.

Hopefully that clears up my confusion, and maybe the confusion of others.